bot/plotting/comparison_generator.py

```diff
@@ -0,0 +1,115 @@
+import pybamm
+import random
+from utils.chemistry_generator import chemistry_generator
+from plotting.plot_graph import plot_graph
+from experiment.experiment_generator import experiment_generator
+
+
+def comparison_generator(
+    number_of_comp,
+    models_for_comp,
+    chemistry,
+    provided_choice=None,
+):
+    params = pybamm.ParameterValues(chemistry=chemistry)
+    parameter_values_for_comp = dict(list(enumerate([params])))
+    comparison_dict = {}
+
+    if number_of_comp == 1:
+        param_list = []
+        diff_params = random.randint(2, 3)
+        for i in range(0, diff_params):
+            param_list.append(params.copy())
+            param_list[i][
+                "Current function [A]"
+            ] = chemistry_generator(
+                chemistry, "Current function [A]"
+            )
+        parameter_values_for_comp = dict(
+            list(enumerate(param_list))
+        )
+
+    choice = random.randint(0, 1)
+    if provided_choice is not None:
+        choice = provided_choice
+
+    if choice == 0:
+        s = pybamm.BatchStudy(
+            models=models_for_comp,
+            parameter_values=parameter_values_for_comp,
+            permutations=True,
+        )
+
+        s.solve([0, 3700])
```

Comment:
If you change the "Current function [A]" you might also want to change the amount of time you are solving for. The default value discharge in around 1 hour, so if e.g. you double the current you might only want to solve for around half an hour. An alternative would be to set it up as an experiment: 

```python3
model = pybamm.lithium_ion.SPM()

parameter_values = pybamm.ParameterValues(chemistry=pybamm.parameter_sets.Marquis2019)
V_low = parameter_values["Lower voltage cut-off [V]"]
I_1C = parameter_values["Current function [A]"]

factor = 2 # randomize this 
I = factor * I
t_end = 1 / factor * 1.1 

experiment = pybamm.Experiment([f"Discharge at {I} A for {t_end} hours or until {V_low}V"])
sim = pybamm.Simulation(model, experiment=experiment, parameter_values=parameter_values)
sim.solve()
```

This is a little slower than just doing a simulation without the experiment, so you could just alter the solve time in what you are already doing.

Reply:
```py
factor = 2 # randomize this 
I = factor * I
t_end = 1 / factor * 1.1 
```
In this script, is `I` the default `"Current function [A]"` value? I think adding this script should make it work

Reply:
Yes, sorry that should be `I = factor * I_1C`

Reply:
where `I_1C` is the default value

Reply:
How should I go about providing the same t_eval for 2-3 different values of `"Current function [A]"` (which is happening here), should I take the max value as a reference or the average values as the reference?

Reply:
So here are you comparing the same model/parameters and just changing the "Current function [A]"? The smallest current will give you the longest time, so use that the calculate the simulation times. The other simulations with a higher current will just stop sooner because they will hit an event (e.g. the low voltage cut-off).

Reply:
Yes, that is what I am doing here. Thanks, I will add this!